core/server/master/src/main/java/alluxio/master/file/replication/ReplicationChecker.java

```diff
@@ -130,7 +131,22 @@ public void heartbeat() throws InterruptedException {
       return;
     }
 
-    TimeUnit.SECONDS.sleep(mQuietPeriodSeconds);
+    final Iterator<Long> jobIterator = mActiveJobToInodeID.keySet().iterator();
+
+    while (jobIterator.hasNext()) {
+      final Long jobId = jobIterator.next();
+      try {
+        final Status jobStatus = mReplicationHandler.getJobStatus(jobId);
+        if (jobStatus.isFinished()) {
+          final long inodeId = mActiveJobToInodeID.get(jobId);
+          mActiveInodeIDs.remove(inodeId);
+          jobIterator.remove();
+        }
+      } catch (IOException e) {
+        e.printStackTrace();
```

Comment:
LOG instead of printstack?

Reply:
There is a DO NOT MERGE here because this was never intended to be merged. Just wanted some feedback.